Dockerfile

```diff
@@ -30,3 +30,22 @@ ENV JAVA_HOME /usr/lib/jvm/default-java
 # Set up baselisk.
 RUN wget https://github.com/bazelbuild/bazelisk/releases/download/v1.10.1/bazelisk-linux-amd64 \
     -O /usr/local/bin/bazelisk && chmod a+x /usr/local/bin/bazelisk
+
+# Rust is needed for authorization logic
+ARG rustup_dir=/usr/local/cargo
+ENV RUSTUP_HOME ${rustup_dir}
+ENV CARGO_HOME ${rustup_dir}
+ENV PATH "${rustup_dir}/bin:${PATH}"
+RUN curl --location https://sh.rustup.rs > /tmp/rustup \
+  && sh /tmp/rustup -y --default-toolchain=none \
+  && chmod a+rwx ${rustup_dir} \
+  && rustup --version
+
+## Set the right version of rust
+ARG RUST_VERSION=nightly-2021-06-30
+RUN rustup toolchain install ${RUST_VERSION} \
+    && rustup default ${RUST_VERSION}
+
+# Souffle is needed to run the tests in authorization logic
+RUN curl -s https://packagecloud.io/install/repositories/souffle-lang/souffle/script.deb.sh | bash
+RUN apt-get update && apt-get install -y souffle
```

Comment:
This doesn't look right. Our project has a copy of Souffle that it already builds and uses as part of the Raksha tests; shouldn't we use the same Souffle version here rather than fetching another version from `apt`?

Reply:
Isn't that version of souffle a part of bazel/BUILD, which I think doesn't put it into the docker environment? I need to run these tests with cargo which does not use bazel. I also ran this without adding souffle here first, and it did not work, so I think it must be neeeded.

Reply:
I think it is OK to install another version of souffle for now. In the current setup now, it is not straightforward to run the tool we built from the command line as it would be in the bazel's cache. We will need to add a `sh_binary` rule to wrap the souffle binary and change how it is invoked from rust here:
https://github.com/google-research/raksha/blob/60fd88a02232b5d2199e0a8d9fa31634137977a5/rust/tools/authorization-logic/src/souffle/souffle_interface.rs#L67

In any case, I agree that we might run into consistency issues. e.g., some feature not working on downloaded version but working on our local version (or vice versa). Hopefully that should not cause major problems. Let us file an issue to add a sh_binary rule for souffle and fix this later.

Reply:
Ah, OK I guess it is fixable by changing how souffle is invoked from within the rust code. Sounds good.